components/skus/browser/rs/lib/src/sdk/credentials/fetch.rs

```diff
@@ -63,17 +96,120 @@ where
             return Err(InternalError::OrderUnpaid.into());
         }
 
-        let mut csprng = OsRng;
 
         for item in order.items {
             match item.credential_type {
+                CredentialType::TimeLimitedV2 => {
+                    let mut num_creds: usize = 0;
+                    if let Some(ref metadata) = order.metadata {
+                        let num_intervals = metadata.num_intervals.unwrap_or(0);
+                        let num_per_interval = metadata.num_per_interval.unwrap_or(0);
+                        num_creds = num_intervals * num_per_interval;
+                    }
+
+                    let blinded_creds: Vec<BlindedToken> = match self
+                        .client
+                        .get_time_limited_v2_creds(&item.id)
+                        .await?
+                    {
+                        Some(item_creds) => {
+                            let initialized: bool =
+                                !item_creds.unblinded_creds.unwrap_or_default().is_empty();
+                            let has_existing_creds: bool = !item_creds.creds.is_empty();
```

Comment:
the naming here is confusing, especially `initialized`. maybe `has_unblinded_creds` or `has_signed_creds`?